dts/client.py

```diff
@@ -104,47 +106,68 @@ def search(self,
-        * `"archived"` means "search only for files that are archived and not staged"
+        * `"unstaged"` means "search only for files that are not staged"
     * offset: a 0-based index from which to start retrieving results (default: 0)
     * limit: if given, the maximum number of results to retrieve
+    * specific: a dictionary mapping database-specific search parameters to their values
 """
+        params = {
+            'database': database,
+            'query':    query,
+        }
         if not self.uri:
             raise RuntimeError('dts.Client: not connected.')
-        if type(database) != str:
+        if not isinstance(query, str):
+            raise RuntimeError('search: missing or invalid query.')
+        if not isinstance(database, str):
             raise TypeError('search: database must be a string.')
-        if type(offset) != int or offset < 0:
-            raise TypeError('search: invalid offset: %s.'%offset)
+        if status:
+            if status not in ['staged', 'unstaged']:
+                raise TypeError(f'search: invalid status: {status}.')
+            params['status'] = status
+        if offset:
+            if not str(limit).isdigit():
+                raise TypeError('search: offset must be numeric')
+            if int(limit) < 0:
+                raise ValueError(f'search: limit must be non-negative')
+            params['offset'] = int(offset)
         if limit:
-            if type(limit) != int:
-                raise TypeError('search: limit must be an int.')
-            elif limit < 1:
-                raise TypeError(f'search: invalid number of retrieved results: {N}')
+            if not str(limit).isdigit():
+                raise TypeError('search: limit must be numeric')
+            if int(limit) < 1:
+                raise ValueError(f'search: limit must be greater than 1')
+            params['limit'] = int(limit)
+        if specific:
+            if not isinstance(specific, dict):
+                raise TypeError('search: specific must be a dict.')
+            params['specific'] = specific
         try:
-            params = {
-                'database': database,
-                'query':    query,
-                'status':   status,
-                'offset':   offset,
-                'limit':    limit,
-            }
-            response = requests.get(url=f'{self.uri}/files', params=params, auth=self.auth)
+            response = requests.post(url=f'{self.uri}/files',
+                                     json=params,
+                                     auth=self.auth)
             response.raise_for_status()
         except HTTPError as http_err:
             logger.error(f'HTTP error occurred: {http_err}')
             return None
+        except requests.exceptions.HTTPError as err:
+            logger.error(f'HTTP error occurred: {err}')
+            return None
```

Comment:
what is the difference between the `urllib`'s `HTTPError` and `requests`'s `HTTPError`? If the action is going to be the same, may as well just do

```py
except (HTTPError, requests.exceptions.HTTPError) as err:
```